wandb/integration/openai/fine_tune.py

```diff
@@ -0,0 +1,338 @@
+import datetime
+import io
+import os
+import json
+import time
+import re
+from pathlib import Path
+
+try:
+    import openai
+    from openai import OpenAI
+except ImportError as e:
+    raise Exception(
+        "Error: `openai` not installed >> This integration requires openai!  To fix, please `pip install --pre openai`"
+    ) from e
+
+try:
+    import numpy as np
+except ImportError as e:
+    raise Exception(
+        "Error: `numpy` not installed >> This integration requires numpy!  To fix, please `pip install numpy`"
+    ) from e
+
+try:
+    import pandas as pd
+except:
+    raise Exception(
+        "Error: `pandas` not installed >> This integration requires pandas!  To fix, please `pip install pandas`"
+    ) from e
+
+import wandb
+from wandb.sdk.lib import telemetry
+
+# from openai import File, FineTune, FineTuningJob
+
+
+class WandbLogger:
+    """
+    Log fine-tunes to [Weights & Biases](https://wandb.me/openai-docs)
+    """
+
+    _wandb_api = None
+    _logged_in = False
+    openai_client = None
+
+    @classmethod
+    def sync(
+        cls,
+        id=None,
+        n_fine_tunes=None,
+        project="OpenAI-Fine-Tune",
+        entity=None,
+        force=False,
+        blocking=True,
+        **kwargs_wandb_init,
+    ):
+        """
+        Sync fine-tunes to Weights & Biases.
+        :param id: The id of the fine-tune (optional)
+        :param n_fine_tunes: Number of most recent fine-tunes to log when an id is not provided. By default, every fine-tune is synced.
+        :param project: Name of the project where you're sending runs. By default, it is "GPT-3".
+        :param entity: Username or team name where you're sending runs. By default, your default entity is used, which is usually your username.
+        :param force: Forces logging and overwrite existing wandb run of the same fine-tune.
+        :param blocking: Waits for the fine-tune to be complete and then log metrics to W&B. By default, it is True.
+        """
+
+        openai_client = OpenAI(
+            api_key=os.environ['OPENAI_API_KEY'],
```

Comment:
This should first check if `openai.api_key` is not none first I think. 

https://platform.openai.com/docs/api-reference/authentication